Servers.yaml

```diff
@@ -0,0 +1,216 @@
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: S3HTMLAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: "*"
+                Resource: "*"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/CloudWatchLogsFullAccess
+      Path: /
+
+  EC2InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /
+      Roles:
+        - !Ref EC2InstanceRole
+
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    DependsOn: EC2InstanceRole
+    Properties:
+      Bucket: !Ref HTMLBucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              AWS: !GetAtt EC2InstanceRole.Arn
+            Action: "*"
+            Resource: 
+              - !Sub "arn:aws:s3:::${HTMLBucket}"
+              - !Sub "arn:aws:s3:::${HTMLBucket}/*"
+
+  LaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    DependsOn: EC2InstanceProfile
+    Properties:
+      LaunchTemplateName: !Ref LaunchTemplateName
+      LaunchTemplateData:
+        InstanceType: t3.micro
+        ImageId: ami-05edb7c94b324f73c  # Amazon Linux 2023
+        KeyName: !Ref KeyPairName
+        SecurityGroupIds:
+          - !ImportValue InstanceSecurityGroupId
+        IamInstanceProfile:
+          Name: !Ref EC2InstanceProfile
+        UserData:
+          Fn::Base64: 
+            !Sub |
+              #!/bin/bash -xe
+              yum update -y
+              yum install -y nginx aws-cli
+
+              aws configure set default.region ${AWS::Region}
+
+              # Create website directory
+              WEBSITE_DIR="/usr/share/nginx/html"
+              rm -rf $WEBSITE_DIR/*
+
+              # Download content from S3
+              aws s3 sync s3://${HTMLBucketName}/ $WEBSITE_DIR/
+
+              # Configure nginx
+              cat > /etc/nginx/conf.d/default.conf << 'EOL'
+              server {
+                  listen 80;
+                  server_name _;
+                  root /usr/share/nginx/html;
+                  index index.html index.htm;
+                  
+                  location / {
+                      try_files $uri $uri/ =404;
+                  }
+              }
+              EOL
+
+              # Set permissions
+              chown -R nginx:nginx $WEBSITE_DIR
+              chmod -R 755 $WEBSITE_DIR
+
+              # Start and enable nginx
+              systemctl enable nginx
+              systemctl start nginx
+
+              # Set up automatic content refresh (every 1 minute)
+              echo "*/1 * * * * root aws s3 sync s3://${HTMLBucketName}/ $WEBSITE_DIR/" > /etc/cron.d/sync-s3-content
```

Comment:
No need to this cron

Reply:
This cronjob is needed because it copies over content from the S3 periodically because i upload the index.html to the S3 bucket after the stack is created. this is the solution the FINALLY solved it.